packages/help-center/src/hooks/use-article-content-click.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, type MouseEvent } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const isThisASupportArticleLink = ( href: string ) =>
+	/wordpress\.com(\/\w\w)?(?=\/support\/)|support\.wordpress\.com/.test( href );
+
+export const useArticleContentClick = ( node: HTMLDivElement | null ) => {
+	const navigate = useNavigate();
+
+	const filters = [
+		/**
+		 * Make support article links open within Help Center.
+		 */
+		{
+			condition: ( target: HTMLElement ) =>
+				target.tagName === 'A' && isThisASupportArticleLink( target.getAttribute( 'href' ) || '' ),
+			action: ( event: MouseEvent< HTMLElement > ) => {
+				event.preventDefault();
+				const target = event.target as HTMLElement;
+				const href = target.getAttribute( 'href' );
+				navigate( `/post?link=${ href }` );
+			},
+		},
+
+		/**
+		 * Fix jump-to-section links.
+		 */
+		{
+			condition: ( target: HTMLElement ) =>
+				target.tagName === 'A' && target.getAttribute( 'href' )?.startsWith( '#' ),
+			action: ( event: MouseEvent< HTMLElement > ) => {
+				const target = event.target as HTMLElement;
+				if ( target.tagName === 'A' ) {
+					event.preventDefault();
+					const href = target.getAttribute( 'href' );
+					node?.querySelector( href as string )?.scrollIntoView();
+				}
+			},
+		},
+	];
+
+	useEffect( () => {
+		node?.addEventListener( 'click', ( event: Event ) => {
```

Comment:
Making this change would probably free you from all the casting below. 

```suggestion
		node?.addEventListener( 'click', ( event: MouseEvent< HTMLElement >  ) => {
```

Reply:
Did you try that in VSCODE? I was stuck in a weird typing hell when I tried to use that and I couldn't get out of it. The types I have now were the only way I could format it to appease TS.

Reply:
I think React's `MouseEvent`was messing with you, here's my take, I also added `useMemo` to fitlers.

```ts
import { useEffect, useMemo } from 'react';
import { useNavigate } from 'react-router-dom';

const isThisASupportArticleLink = ( href: string ) =>
	/wordpress\.com(\/\w\w)?(?=\/support\/)|support\.wordpress\.com/.test( href );

export const useArticleContentClick = ( node: HTMLDivElement | null ) => {
	const navigate = useNavigate();

	const filters = useMemo(
		() => [
			/**
			 * Make support article links open within Help Center.
			 */
			{
				condition: ( target: HTMLElement ) =>
					target.tagName === 'A' &&
					isThisASupportArticleLink( target.getAttribute( 'href' ) || '' ),
				action: ( event: MouseEvent ) => {
					event.preventDefault();
					const target = event.target as HTMLElement;
					const href = target.getAttribute( 'href' );
					navigate( `/post?link=${ href }` );
				},
			},

			/**
			 * Fix jump-to-section links.
			 */
			{
				condition: ( target: HTMLElement ) =>
					target.tagName === 'A' && target.getAttribute( 'href' )?.startsWith( '#' ),
				action: ( event: MouseEvent ) => {
					const target = event.target as HTMLElement;
					if ( target.tagName === 'A' ) {
						event.preventDefault();
						const href = target.getAttribute( 'href' );
						node?.querySelector( href as string )?.scrollIntoView();
					}
				},
			},
		],
		[ node, navigate ]
	);

	useEffect( () => {
		node?.addEventListener( 'click', ( event: MouseEvent ) => {
			const target = event.target as HTMLElement;

			filters.forEach( ( { condition, action } ) => {
				if ( condition( target ) ) {
					action( event );
				}
			} );
		} );
	}, [ node, filters ] );
};
```